### PR TITLE
feat: added confirm and get handler for last used payment

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

`confirmWithLastUsedPaymentMethod` & `getCustomerLastUsedPaymentMethodData` these 2 headless functions added.

## How did you test it?

Via headless function integration.

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/orca-loader/PaymentSessionMethods.res [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+
+  if temp1 === temp2 {
+    0.
+  } else if temp1 > temp2 {
+    -1.
+  } else {
+    1.
+  }
+}
+
```
Comment (PritishBudhiraja) [inline src/orca-loader/PaymentSessionMethods.res:14]: Will move this function out when this PR will be merged - https://github.com/juspay/hyperswitch-web/pull/427

**Review by ArushKapoorJuspay — DISMISSED** [review]

File: src/orca-loader/PaymentSessionMethods.res [hunk 2]
```diff
@@ -17,140 +31,160 @@ let getCustomerSavedPaymentMethods = (
...
+    let customerPaymentMethods = customerDetailsArray->Array.filter(customerPaymentMethod => {
+      customerPaymentMethod
+      ->JSON.Decode.object
+      ->Option.flatMap(x => x->Dict.get("default_payment_method_set"))
+      ->Option.flatMap(JSON.Decode.bool)
+      ->Option.getOr(false)
+    })
+
+    let paymentNotExist = (~message="There is no customer default saved payment method data") =>
```
Comment (vsrivatsa-juspay) [inline src/orca-loader/PaymentSessionMethods.res:49]: /
```suggestion
    let paymentNotExist = (~message="There is no default saved payment method data for this customer") =>
```

File: src/orca-loader/PaymentSessionMethods.res [hunk 3]
```diff
@@ -17,140 +31,160 @@ let getCustomerSavedPaymentMethods = (
...
-          ~isPaymentSession=true,
-          (),
-        )
+    let getCustomerLastPaymentMethodUsedData = () => {
+      let customerPaymentMethodsCopy = customerDetailsArray->Array.copy
+      customerPaymentMethodsCopy->Array.sort(sortFunctions)
+
+      switch customerPaymentMethodsCopy->Array.get(0) {
+      | Some(customerLastPaymentUsed) => customerLastPaymentUsed
+      | None => paymentNotExist(~message="No recent payments found.")
```
[the remark this answers is not in the document]
Comment (vsrivatsa-juspay) [reply src/orca-loader/PaymentSessionMethods.res:72]: Also if you are adding a period here, add it in the message above in the same file as well

File: src/orca-loader/PaymentSessionMethods.res [hunk 4]
```diff
@@ -17,140 +31,160 @@ let getCustomerSavedPaymentMethods = (
...
+      let customerPaymentMethodsCopy = customerDetailsArray->Array.copy
+      customerPaymentMethodsCopy->Array.sort(sortFunctions)
+
+      switch customerPaymentMethodsCopy->Array.get(0) {
+      | Some(customerLastPaymentUsed) =>
+        confirmCallForParticularPaymentObject(
+          ~paymentMethodObject=customerLastPaymentUsed,
+          ~payload,
+        )
+      | None => paymentNotExist(~message="No recent payments found.")->resolve
```
Comment (vsrivatsa-juspay) [inline src/orca-loader/PaymentSessionMethods.res:176]:
```suggestion
      | None => paymentNotExist(~message="No recent payments found for this customer")->resolve
```

**Review by prafulkoppalkar — APPROVED** [review]

**Review by ArushKapoorJuspay — DISMISSED** [review]

**Review by ArushKapoorJuspay — DISMISSED** [review]

**Review by ArushKapoorJuspay — APPROVED** [review]

**Review by prafulkoppalkar — APPROVED** [review]

**Pull request merged**